### DISCUSSION TITLE
How to get access to ECS Container credentials through JS SDK? (details in thread)

### DISCUSSION THREAD

**Discussion by ReadyElbow** [aws/aws-sdk-js-v3#3690] [Q&A]
I have a Node JS application running within a ECS Container and I am looking to use the attached IAM Task Role to connect to an S3 bucket. 

From what I have gathered online and through the docs, this would be automatically picked up if I were to use the S3 Client as it would use the default provider credential chain. 

However, as it stands I am currently using the MinIO Client which requires specific definitions on the Access Key, Session Key, Secret Key passed to it; essentially I am trying to gain _direct_ access to the credential values as an object. 
E.g.
`
var s3Client = new Minio.Client({
    endPoint:  's3.amazonaws.com',
    accessKey: fetchedCredential,
    secretKey: fetchedCredential,
    sessionToken: fetchedCredential,
})`

I have attempted to use the function `fromNodeProviderChain()` detailed here [documentation
](https://www.npmjs.com/package/@aws-sdk/credential-providers#fromcontainermetadata-and-frominstancemetadata) which I hope will return a credential object with a list of the keys I need.

Now I have not configured this function further as the documentation is confusing e.g.

```
const credentialProvider = fromNodeProviderChain({
  //...any input of fromEnv(), fromSSO(), fromTokenFile(), fromIni(),
  // fromProcess(), fromInstanceMetadata(), fromContainerMetadata()
  // Optional. Custom STS client configurations overriding the default ones.
  clientConfig: { region },
});
```
Is anyone able to help me identify what command I can run from the SDK that will give me direct access to a Credential Object using the default lookup chain?

Note: we are looking to move away from the MinIO client but we would like to start with this first issue.

**Comment by aBurmeseDev** [comment]
Hi @ReadyElbow - thanks for reaching out and apology for not getting here sooner. 

I'd recommend checking out our [official docs](https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide/setting-credentials-node.html) on setting credentials with Node.js and using credential provider chain. As far as connecting to S3 client, this's what you could do:
```ts
import { S3Client, ListBucketsCommand } from "@aws-sdk/client-s3";

const S3_Config = {
    region: "region",
    credentials: {
    accessKeyId:  ACCESS_KEY_ID,
    secretAccessKey: ACCESS_SECRET_KEY,
    },
};
const s3_client = new S3Client(S3_Config);
const input = {}
const _command = new ListBucketsCommand(input);
const response = await s3_client.send(_command);
console.log(response)
```
Hope it helps,
John
